### PR TITLE
heaptrack doc

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
**Summary of changes**
Changes introduced in this pull request:
- added heaptrack quick start to documents

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/1817

**Other information and links**

### CODE REVIEW

**Review by lemmih — APPROVED** [review]
With pictures and everything!

File: documentation/developer_documentation/memory-analysis.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+```
+heaptrack --analyze heaptrack.forest.12345.gz
+```
+
+
+
+### Summary
+Here we can see memory usage overview. Keep in mind that *leaks* here are not necessarily leaks - it's just memory that hasn't been yet freed. A global cache would always show as a leak.
+
+While most of the potential *culpripts* are not necessarily interesting (e.g. `alloc::*`) because even a `String` constructor calls them, we immediately see that among specific ones, it's the `rocksdb` that gets into the spotlight.
```
Comment (elmattic) [inline documentation/developer_documentation/memory-analysis.md:43]: Typo on *culprits*.

File: documentation/developer_documentation/memory-analysis.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+### Temporary allocations
+Shows the number of temporary allocations over time. Temporary allocation is an allocation followed by its deallocation, i.e. there are no other allocations in-between.
+
+### Sizes
+
+This tab will show you the allocation sizes during runtime and their frequency. If you hover over a bar you will see that e.g. `LZ4_createStream` (most likely used by `rocksdb`) made 5,624,180 allocations, total 92.3G, on average 14.4kB per allocation.
+
+![sizes](heaptrack/sizes.png)
+
+### Miscallenous
```
Comment (elmattic) [inline documentation/developer_documentation/memory-analysis.md:82]: Typo on Miscellaneous.

File: documentation/developer_documentation/memory-analysis.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+- Keep in mind that running Forest *with* heaptrack gives a non-negligible memory and CPU overhead. You may not be able to run mainnet node on a 16G machine even if normally it would be fine.
+- Optimizations may play tricks on the developer, e.g. inlining functions so they won't even appear in your trace. If you think a particular method should have been called but for mysterious reasons it does not appear in the analysis, you may want to put `#[inline(never)]` on top of it. Analyzing a debug build may also be useful, but depending on where the leak happens, it may be too slow.
+- There is a lot of noise coming from dependencies and standard library. It's useful to mentally filter them out a bit and focus on the biggest culprits in Forest methods. Flamegraph and caller/calee view are the most useful for this.
```
Comment (elmattic) [inline documentation/developer_documentation/memory-analysis.md:86]: Typo on callee.

**Review by elmattic — CHANGES_REQUESTED** [review]
A part a few typos, looks like a really detailed how-to for `heaptrack` 👍.

Possibly will have something to add once I start using this tool but for now looks good to merge.

**Review by elmattic — APPROVED** [review]

**Pull request merged**